metrics-datadog/src/main/java/org/coursera/metrics/datadog/RegexMetricFilter.java

```diff
@@ -0,0 +1,46 @@
+package org.coursera.metrics.datadog;
+
+import com.codahale.metrics.Metric;
+import com.codahale.metrics.MetricFilter;
+import com.google.common.collect.ImmutableSet;
+
+import java.util.regex.Pattern;
+
+public class RegexMetricFilter implements MetricFilter {
```

Comment:
@ToadJam sorry for dropping out on this.  My initial PR comment was around whether this filter should be included in either module under metrics-datadog given that it isn't specific to working with datadog. If I had to choose, I'd leave it in the `dropwizard-metrics-datadog` module, since their factory allows for configuration of a `MetricFilter`. @tianx2 Can chime in if he feels otherwise.

If the decision is to include it in the metrics-datadog module, I'm also hesitant to introduce the Guava dep. It's a heavyweight dep (2.2 MB) and has been had incompatibility issues in the past.  Perhaps you can just use a JDK `Set` implementation?